sdk_blueprints/Gretel_101_Blueprint.ipynb

```diff
@@ -0,0 +1 @@
ution_count":null,"metadata":{"id":"cW3VKpyPvm6W"},"outputs":[],"source":["import pandas as pd\n","\n","# explore the data using pandas\n","df = pd.read_csv(dataset)\n","df.head()"]},{"cell_type":"markdown","metadata":{"id":"SwROZthrvXil"},"source":["## 🏋️‍♂️ Train a generative model\n","\n","- The [tabular-actgan](https://github.com/gretelai/gretel-blueprints/blob/main/config_templates/gretel/synthetics/tabular-actgan.yml) base config tells Gretel which model to train and how to configure it.\n","\n","- You can replace `tabular-actgan` with the path to a custom config or select any of the tabular configs [listed here](https://github.com/gretelai/gretel-blueprints/tree/main/config_templates/gretel/synthetics).\n","\n","- The training data is passed in using the `data_source` argument. Its type can be a file path or `DataFrame`.\n","\n","- **Tip:** Click the printed Console URL to monitor your job's progress in the Gretel Console."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"i89eGZwIxSCW"},"outputs":[],"source":["trained = gretel.submit_train(\"tabular-actgan\", data_source=dataset)"]},{"cell_type":"markdown","metadata":{"id":"eljkfb8jb_hK"},"source":["## 🧐 Evaluate the synthetic data quality\n","\n","- Gretel automatically creates a [synthetic data quality report](https://docs.gretel.ai/reference/evaluate/synthetic-data-quality-report) for each model you train.\n","\n","- The training results object returned by `submit_train` has a `GretelReport` attribute for viewing the quality report.\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"bNZqhFPOclrV"},"outputs":[],"source":["# view the quality scores\n","print(trained.report)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"3QMiP7lKecE5"},"outputs":[],"source":["# display the full report within this notebook\n","trained.report.display_in_notebook()"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"2dHuQT_cuIno"},"outputs":[],"source":["# inspect the synthetic data used to create the report\n","df_synth_report = trained.fetch_report_synthetic_data()\n","df_synth_report.head()"]},{"cell_type":"markdown","metadata":{"id":"ZIeY7TczxvDV"},"source":["## 🤖 Generate synthetic data\n","\n","- The `model_id` argument can be the ID for any trained model within the current project.\n","\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"J6XZUuR2eguX"},"outputs":[],"source":["generated = gretel.submit_generate(trained.model_id, num_records=1000)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"-_do0Kvvunv2"},"outputs":[],"source":["# inspect the generated synthetic data\n","generated.synthetic_data.head()"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"TZTK8H6f7wdS"},"outputs":[],"source":[]}],"metadata":{"colab":{"authorship_tag":"ABX9TyPpR+I4NsfIgheAovckx+Wy","provenance":[]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}
```

Comment:
nit: Could you add a bullet point under 'Train a Generative Model' that mentions the user has the option to set 'wait=False' and then able to monitor training progress by running `trained.wait_for_completion()`
And similar for `submit_generate`

Reply:
nit: delete empty cell at the end

Reply:
nit: do we need to explain why `validate=True` during configuration?

Reply:
Thanks @mvansegbroeck!

I addressed the second two comments. I tried adding the note about `wait=False`, but there's actually a fair amount to say about that option. I'm thinking the docs will be a good place for that. It's also somewhat advanced, since you have to run code asynchronously with the job. WDYT?

For the docs, I'm thinking something like this: 

## ⏳ To wait or not to wait?

If you do not want to wait for a job to complete, you can set `wait=False` when calling  `submit_train` or `submit_generate`. In this case, the method will return immediately after the job starts. 

Some things to know if you use this option:

- You can still monitor the job progress in the Gretel Console.

- To pause your code and monitor the job progress locally, call the `wait_for_completion` method of the returned object.

- You can check the job status using the `job_status` attribute of the returned object.

- If you are not waiting when the job completes, you must call the `refresh` method of the returned object to fetch the job results.

Reply:
Ok, make sense. Maybe good to drop a link to the docs in the notebook as well - but maybe this has to be done afterwards?

Reply:
First I have to write the docs 🙃 

Still need to decide how we want to integrate them into GitBook vs Sphinx